code/modules/atmospherics/machinery/components/fusion/hfr_parts.dm

```diff
@@ -322,49 +322,44 @@
 			var/heating_conductor = params["heating_conductor"]
 			if(text2num(heating_conductor) != null)
 				heating_conductor = text2num(heating_conductor)
-				. = TRUE
-			if(.)
 				connected_core.heating_conductor = clamp(heating_conductor, 50, 500)
+				. = TRUE
 		if("magnetic_constrictor")
 			var/magnetic_constrictor = params["magnetic_constrictor"]
 			if(text2num(magnetic_constrictor) != null)
 				magnetic_constrictor = text2num(magnetic_constrictor)
-				. = TRUE
-			if(.)
 				connected_core.magnetic_constrictor = clamp(magnetic_constrictor, 50, 1000)
+				. = TRUE
 		if("fuel_injection_rate")
 			var/fuel_injection_rate = params["fuel_injection_rate"]
 			if(text2num(fuel_injection_rate) != null)
 				fuel_injection_rate = text2num(fuel_injection_rate)
-				. = TRUE
-			if(.)
 				connected_core.fuel_injection_rate = clamp(fuel_injection_rate, 5, 1500)
+				. = TRUE
 		if("moderator_injection_rate")
 			var/moderator_injection_rate = params["moderator_injection_rate"]
 			if(text2num(moderator_injection_rate) != null)
 				moderator_injection_rate = text2num(moderator_injection_rate)
-				. = TRUE
-			if(.)
 				connected_core.moderator_injection_rate = clamp(moderator_injection_rate, 5, 1500)
+				. = TRUE
 		if("current_damper")
 			var/current_damper = params["current_damper"]
 			if(text2num(current_damper) != null)
 				current_damper = text2num(current_damper)
-				. = TRUE
-			if(.)
 				connected_core.current_damper = clamp(current_damper, 0, 1000)
+				. = TRUE
 		if("waste_remove")
 			connected_core.waste_remove = !connected_core.waste_remove
 			. = TRUE
 		if("filter")
-			connected_core.filter_type = null
-			var/filter_name = "nothing"
-			var/gas = gas_id2path(params["mode"])
-			if(gas in GLOB.meta_gas_info)
-				connected_core.filter_type = gas
-				filter_name = GLOB.meta_gas_info[gas][META_GAS_NAME]
-			investigate_log("was set to filter [filter_name] by [key_name(usr)]", INVESTIGATE_ATMOS)
+			connected_core.moderator_scrubbing ^= gas_id2path(params["mode"])
```

Comment:
could someone break something by putting whatever they wanted in here?

Reply:
no, because you can only click on specific gases, just like the portable scrubber is, there isn't a way to break this afaik